primitives/src/lib.rs

```diff
@@ -164,7 +164,7 @@ impl<AccountId> Default for EraValidators<AccountId> {
 
 #[derive(Encode, Decode, PartialEq, Eq, Debug)]
 pub enum ApiError {
-    DecodeKey,
+    MissingAuthoritiesForNextSession,
```

Comment:
Not entirely sure but this change might be problematic.

Because of change in this type, the runtime Api changes, and the new host knows only the new runtime Api. So after an update, the host will use the types for the new Api to call the old on-chain Api. This might sound confusing, and rightfully so, so I'm happy to explain on slack or during a call. 

What's bad is that this will compile fine. Even during runtime this might be fine, because we will be lucky and things will deserialize correctly. But all the time we will be deserializing to a different type than the one that the data was serialized from.